File: benchmarks/beginner/B3-array-sum/javascript/main.js
function main() {
    const N = 10_000_000;
    const RUNS = 10;
    const EXPECTED = 50000005000000;
    const timings = [];

    console.log("=== BENCHMARK: B3 - Array Sum ===");
    console.log("Language: JavaScript");
    console.log(`Runtime:  Node.js ${process.version}`);
    console.log();

    let result = 0;
    for (let i = 1; i <= RUNS; i++) {
        const start = process.hrtime.bigint();
        const arr = new Array(N);
        for (let j = 0; j < N; j++) {
            arr[j] = j + 1;
        }
        let s = 0;
        for (let j = 0; j < N; j++) {
            s += arr[j];
        }
        result = s;
        const elapsed = Number(process.hrtime.bigint() - start) / 1e6; // ms
        timings.push(elapsed);
        console.log(`Run ${String(i).padStart(2)}: ${elapsed.toFixed(2)} ms`);
    }

    console.log();
    const min = Math.min(...timings);
    const max = Math.max(...timings);
    const avg = timings.reduce((a, b) => a + b, 0) / timings.length;
    const variance = timings.reduce((a, t) => a + (t - avg) ** 2, 0) / timings.length;
    const stddev = Math.sqrt(variance);

    console.log(`Min:    ${min.toFixed(2)} ms`);
    console.log(`Avg:    ${avg.toFixed(2)} ms`);
    console.log(`Max:    ${max.toFixed(2)} ms`);
    console.log(`StdDev: ${stddev.toFixed(2)} ms`);
    console.log();
    const status = result === EXPECTED ? "✓" : "✗";
    console.log(`Verification: ${result} ${status}`);
}

main();

File: benchmarks/beginner/B3-array-sum/rust/main.rs
use std::hint::black_box;
use std::time::Instant;

fn main() {
    const N: usize = 10_000_000;
    const RUNS: usize = 10;
    const EXPECTED: i64 = 50000005000000;
    let mut timings = [0.0f64; RUNS];

    println!("=== BENCHMARK: B3 - Array Sum ===");
    println!("Language: Rust");
    println!("Runtime:  rustc (native)");
    println!();

    let mut result: i64 = 0;
    for i in 0..RUNS {
        let start = Instant::now();
        let arr: Vec<i64> = (1..=N as i64).collect();
        let s: i64 = black_box(&arr).iter().sum();
        result = black_box(s);
        let elapsed = start.elapsed().as_secs_f64() * 1000.0; // ms
        timings[i] = elapsed;
        println!("Run {:2}: {:.2} ms", i + 1, elapsed);
    }

    println!();
    let min = timings.iter().cloned().fold(f64::INFINITY, f64::min);
    let max = timings.iter().cloned().fold(f64::NEG_INFINITY, f64::max);
    let avg = timings.iter().sum::<f64>() / RUNS as f64;
    let variance = timings.iter().map(|t| (t - avg).powi(2)).sum::<f64>() / RUNS as f64;
    let stddev = variance.sqrt();

    println!("Min:    {:.2} ms", min);
    println!("Avg:    {:.2} ms", avg);
    println!("Max:    {:.2} ms", max);
    println!("StdDev: {:.2} ms", stddev);
    println!();
    let status = if result == EXPECTED { "✓" } else { "✗" };
    println!("Verification: {} {}", result, status);
}
